src/MetricsProvider.ts

```diff
@@ -19,8 +21,10 @@ export default class MetricsProvider {
 
   private sessionStarted: boolean = false
   private readonly _consentGranted: Set<consentTypes> = new Set()
+  private readonly metricsService: T
 
-  constructor ({ autoTrack = true, url = COUNTLY_API_URL, appKey }: MetricsProviderConstructorOptions) {
+  constructor ({ autoTrack = true, url = COUNTLY_API_URL, appKey, metricsService }: MetricsProviderConstructorOptions<T>) {
```

Comment:
Usually the lint will fix this when it needs breaking out. We shouldn't need to worry about style things like this since our build does it (should do it?) for us.

And if we want it to be broken up earlier, we should update rules so that it happens automatically.